### DISCUSSION TITLE
Misleading example in .NET 5 design document

### DISCUSSION THREAD

**Discussion by Flavien** [dotnet/designs#258] [General]
There seems to be a confusing example (potentially wrong?) in the document about [Target Framework Names in .NET 5](https://github.com/dotnet/designs/blob/main/accepted/2020/net5/net5.md). This is the example about using TFMs to [vary implementation](https://github.com/dotnet/designs/blob/main/accepted/2020/net5/net5.md#vary-implementation).

It says:

> Ida is working on a Xamarin Forms application that supports Android, iOS, and Windows. Her application needs GPS information, but only a very limited set. Since there is no portable GPS API, **she writes her own little abstraction library using multi-targeting**.
>
> By doing so, she's able to encapsulate the GPS access **without having to multi-target her entire application, just this one area**.

So I imagine there is a `XamarinApplication` project targeting `net5.0` (it says "without having to multi-target her entire application") and a `XamarinGps` library targeting `net5.0-ios;net5.0-android;net5.0-windows;net5.0` of which the implementation is outlined in the example.

The problem is that since `XamarinApplication` is not using multitargeting and targets just `net5.0`, the version of `XamarinGps` that will be used is the `net5.0` one, and that version has no implementation (just `throw new PlatformNotSupportedException()`).

Unless I misunderstand something, it will not work unless `XamarinApplication` also targets `net5.0-ios;net5.0-android;net5.0-windows` (in which case the platform specific implementations will be used).

If Ida wants to avoid multitargeting in `XamarinApplication`, it sounds like she should be using runtime identifiers, not TFMs.

Am I missing something or is the example misleading?

cc: @terrajobst since he's the author.

**Comment by akoeplinger** [comment]
Yes I think you're right, the main application would need to multi-target as well.
